### PR TITLE
Advanced search panel

### PR DESCRIPTION
**Pull request by Max-Levitskiy**

### CODE REVIEW

File: web/src/app/components/resources/list/list.component.ts [hunk 1]
```diff
@@ -31,27 +35,33 @@ export class ListComponent implements AfterViewInit {
   total = 0;
   isLoadingResults = true;
   private query: string;
+  form$: Observable<FormGroupState<SearchParams>>;
+  private controls: FormGroupControls<SearchParams>;
 
-  constructor(private resourceService: ResourceService) {
+  constructor(private resourceService: ResourceService, private store: Store<State>) {
+    this.form$ = this.store.select('searchForm');
```
Comment (helloilya) [inline web/src/app/components/resources/list/list.component.ts:42]: Better to move it into `ngOnInit()`.

File: web/src/app/components/resources/list/list.component.ts [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
 import {QueryParams, SortParam} from '../../../model/es/es-query';
+import {Store} from '@ngrx/store';
+import {FormGroupControls, FormGroupState} from 'ngrx-forms';
+import {State} from '../../../reducers';
+import {SearchParams} from '../../../forms/search-drawer-form';
 
 @Component({
   selector: 'app-list',
   templateUrl: './list.component.html',
   styleUrls: ['./list.component.scss']
```
Comment (helloilya) [inline web/src/app/components/resources/list/list.component.ts:18]: Always use `ChangeDetectionStrategy.OnPush` strategy.

File: web/src/app/components/resources/list/list.component.ts [hunk 3]
```diff
@@ -25,33 +30,45 @@ export class ListComponent implements AfterViewInit {
...
 
-    merge(this.paginator.page, this.sort.sortChange, this.filterEmitter)
+  ngAfterViewInit(): void {
+    this.formSubscription = this.form$.subscribe(form => this.controls = form.controls);
+    this.resources = merge(this.paginator.page, this.sort.sortChange, this.form$)
       .pipe(
         startWith({}),
-        switchMap(() => {
+        switchMap((value) => {
           this.isLoadingResults = true;
```
Comment (helloilya) [inline web/src/app/components/resources/list/list.component.ts:58]: You can use `tap` operator for `this.isLoadingResults`.

File: web/src/app/components/resources/list/list.component.ts [hunk 4]
```diff
@@ -25,33 +30,45 @@ export class ListComponent implements AfterViewInit {
...
           const queryParams: QueryParams = {
```
Comment (helloilya) [inline web/src/app/components/resources/list/list.component.ts:59]: I'd move preparation `queryParams` for `resourceService` into `map`.

File: web/src/app/components/resources/list/list.component.ts [hunk 5]
```diff
@@ -25,33 +30,45 @@ export class ListComponent implements AfterViewInit {
   ];
   @ViewChild(MatPaginator) paginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
-  filterEmitter = new EventEmitter();
 
-  resources: EsHits<Resource>[] = [];
+  resources: Observable<EsHits<Resource>[]>;
```
Comment (helloilya) [inline web/src/app/components/resources/list/list.component.ts:34]: `$` suffix is used to indicate observable.

File: web/src/app/components/resources/list/list.component.ts [hunk 6]
```diff
@@ -25,33 +29,45 @@ export class ListComponent implements AfterViewInit {
...
+  constructor(private resourceService: ResourceService, private store: Store<State>) {
   }
 
-  ngAfterViewInit(): void {
+  ngOnInit(): void {
+    this.form$ = this.store.select('searchForm');
+  }
+
+  ngOnDestroy(): void {
+    this.formSubscription?.unsubscribe();
```
[the remark this answers is not in the document]
Comment (Max-Levitskiy) [reply web/src/app/components/resources/list/list.component.ts:49]: ok :)

File: web/src/app/components/resources/list/list.component.ts [hunk 7]
```diff
@@ -25,33 +29,45 @@ export class ListComponent implements AfterViewInit {
...
+  }
 
-    merge(this.paginator.page, this.sort.sortChange, this.filterEmitter)
+  ngAfterViewInit(): void {
+    this.formSubscription = this.form$.subscribe(form => this.controls = form.controls);
+    merge(this.paginator.page, this.sort.sortChange, this.form$)
       .pipe(
         startWith({}),
-        switchMap(() => {
+        switchMap((value) => {
```
[the remark this answers is not in the document]
Comment (Max-Levitskiy) [reply web/src/app/components/resources/list/list.component.ts:57]: Agree. :)

File: web/src/app/components/resources/list/list.component.html [hunk 8]
```diff
@@ -2,46 +2,51 @@
...
-      <th mat-header-cell *matHeaderCellDef>Tags</th>
-      <td mat-cell *matCellDef="let element">
-        <span *ngFor="let tag of element._source.Tags | keyvalue">{{tag.key}}:{{tag.value}} </span>
-      </td>
-    </ng-container>
+    <mat-form-field>
+      <mat-label>Filter</mat-label>
+      <input matInput [ngrxFormControlState]="(form$ | async).controls.query" placeholder="Filter"/>
+    </mat-form-field>
+    <br/>
```
[the remark this answers is not in the document]
Comment (Max-Levitskiy) [reply web/src/app/components/resources/list/list.component.html:12]: ok, I'll delete it.

File: web/src/app/components/resources/list/list.component.ts (hunk 3 above)
Comment (Max-Levitskiy) [reply web/src/app/components/resources/list/list.component.ts:58]: done

File: web/src/app/components/resources/list/list.component.ts (hunk 4 above)
Comment (Max-Levitskiy) [reply web/src/app/components/resources/list/list.component.ts:59]: done

File: web/src/app/components/resources/list/list.component.ts (hunk 5 above)
Comment (Max-Levitskiy) [reply web/src/app/components/resources/list/list.component.ts:34]: done

**Pull request merged**